packages/edit-site/src/hooks/use-theme-style-variations/use-theme-style-variations-by-property.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { mergeBaseAndUserConfigs } from '../../components/global-styles/global-styles-provider';
+import cloneDeep from '../../utils/clone-deep';
+
+/**
+ * Returns a new object with only the properties specified in `properties`.
+ *
+ * @param {Object} object   The object to filter
+ * @param {Object} property The property to filter by
+ * @return {Object} The merged object.
+ */
+export const filterObjectByProperty = ( object, property ) => {
+	if ( ! object ) {
+		return {};
+	}
+
+	const newObject = {};
+	Object.keys( object ).forEach( ( key ) => {
+		if ( key === property ) {
+			newObject[ key ] = object[ key ];
+		} else if ( typeof object[ key ] === 'object' ) {
```

Comment:
Do we need to guard against `null` here, too? Or do a truthy check + `typeof object[ key ] === 'object'` ?

Reply:
Good point. Wouldn't hurt, I've only tested with object keys so far, e.g., 'typography`, `layout` etc

Reply:
Actually, it's only checking for existing keys in the object:

`Object.keys( object ).forEach( ( key )`

Reply:
Oh, sorry, it's a little nuanced this one, so I might've gotten it wrong. When we're looking at `object[ key ]`, `key` is always a value, but `object[ key ]` could be a `null` value, and `null` unfortunately evaluates to an object type in the `typeof object[ key ] === 'object'` check.

For a more concrete example, what happens if there is a legal value for a property in `theme.json` of `null`? One example would be `settings.spacing.blockGap`, though it mightn't be a particularly useful example 😄

Reply:
> if there is a legal value for a property in theme.json of null? One example would be settings.spacing.blockGap

I'm pretty sure it'll be assigned whatever the value, so long as the property keys match.

```js
		if ( key === property ) {
			newObject[ key ] = object[ key ];

```

I'll add a test to cover `null` as well, though I did but missed it. 👍🏻

Reply:
Okay, great. Thanks for confirming!

Reply:
I appreciate the line of inquiry!!! 🙇🏻